fiwareskuld/create_users.py

```diff
@@ -0,0 +1,181 @@
+        self.keystone = clients.get_keystoneclientv3()
+
+    def add_domain_user_role(self, user, role, domain):
+        """ It adds a role to a user.
+        :param user: the user
+        :param role: the role to add
+        :param domain: the domain
+        :return:
+        """
+        manager = self.keystone.roles
+        return manager.grant(role, user=user, domain=domain)
+
+    def update_domain_to_role(self, user, role_name, date_now=None, duration=100):
+        """
+        It updates the domain to the role
+        :param user: the user
+        :param role_name: the role
+        :param duration: the duration for community
+        :return:
+        """
+        role = self.keystone.roles.find(name=role_name)
+
+        if date_now:
+            date_out = str(date_now)
+        else:
+            date_out = str(datetime.date.today())
+
+        if self.role_name == "community":
+            self.keystone.users.update(user, community_started_at=date_out, community_duration=duration)
+
+        if self.role_name == "trial":
+            self.keystone.users.update(user, trial_started_at=date_out)
+
+        self.add_domain_user_role(
+            user=user,
+            role=role.id,
+            domain='default')
+
+    def create_user(self, user_name, password, role_name, date_now=None):
+        """ It creates a user
+        :return:
+        """
+        self.user_name = user_name
+        self.password = password
+        self.tenant_name = user_name
+        self.role_name = role_name
+        users = self.keystone.users.list(username=self.user_name)
+        if len(users) > 0:
+            raise Exception(403, "User already exist")
+        user = self.keystone.user_registration.users.register_user(
+                self.user_name,
+                domain="default",
+                password=self.password,
+                username=self.user_name)
+
+        self.keystone.user_registration.users.activate_user(
+                user=user.id,
+                activation_key=user.activation_key)
+        users = self.keystone.users.list(username=self.user_name)
+
+        self.update_domain_to_role(users[0], self.role_name, date_now)
+        self.update_quota(users[0], self.role_name)
+        return users[0]
+
+    def update_user(self, user, date=None):
+        """
+        It updates the users.
+        :param user: the user
+        :param date: the date
+        :return: nothing()
+        """
+        self.update_domain_to_role(user, self.role_name, date)
+        self.update_quota(user, self.role_name)
+
+    def delete_user(self, user):
+        """
+        It deletes the user
+        :param user: the user to be deleted.
+        :return: nothing()
+        """
+        d = self.keystone.users.list(username=user.username)
+        if d or len(d) > 0:
+            projects = self.keystone.projects.list(user=d[0])
+            for pro in projects:
+                self.keystone.projects.delete(pro)
```

Comment:
I suppose that you are deleting the user from the projects, maybe it would be nice that you put a comment to clarify it.